### PR TITLE
List and delete instances

### PR DESCRIPTION
**Pull request by kaklakariada**

### CODE REVIEW

File: extension/src/extension.ts [hunk 1]
```diff
@@ -37,16 +39,16 @@ export function createExtension(): ExasolExtension {
         findInstallations(_context: Context, metadata: ExaMetadata): Installation[] {
             return findInstallations(metadata.allScripts.rows);
         },
-        findInstances(_context: Context, _installation: Installation): Instance[] {
-            return [];
+        findInstances(context: Context, version: string): Instance[] {
+            return findInstances(context);
```
Comment (ckunki) [inline extension/src/extension.ts:43]: Version needs to be evaluated in future, right?

File: src/test/java/com/exasol/adapter/document/files/extension/ExtensionIT.java [hunk 2]
```diff
@@ -200,6 +221,23 @@ void createInstanceWithSingleQuote() {
                 .row("Quoted'schema", "SYS", "EXA_EXTENSIONS.S3_FILES_ADAPTER", not(emptyOrNullString())).matches());
     }
 
+    @Test
+    void deleteNonExistingInstance() {
+        assertDoesNotThrow(() -> setup.client().deleteInstance("no-such-instance"));
```
Comment (ckunki) [inline src/test/java/com/exasol/adapter/document/files/extension/ExtensionIT.java:226]: Does this behavior match the requirements?

File: src/test/java/com/exasol/adapter/document/files/extension/ExtensionManagerClient.java [hunk 3]
```diff
@@ -51,38 +51,20 @@ public List<ExtensionsResponseExtension> getExtensions() {
...
-        if (extension.getInstallableVersions().isEmpty()) {
-            throw new IllegalStateException(
-                    "Expected at least one installable version for extensions " + extension.getId());
-        }
-        final String version = extension.getInstallableVersions().get(0);
-        LOGGER.fine(() -> "Installing extension " + extension.getId() + " in version " + version);
-        install(extension.getId(), version);
+        final ExtensionInfo extension = getExtensionInfo();
+        LOGGER.fine(() -> "Installing extension " + extension.getId() + " in version " + extension.getCurrentVersion());
+        install(extension.getId(), extension.getCurrentVersion());
```
Comment (ckunki) [inline src/test/java/com/exasol/adapter/document/files/extension/ExtensionManagerClient.java:67]:
```suggestion
        installExtension(extension.getCurrentVersion());
```

File: src/test/java/com/exasol/adapter/document/files/extension/ExtensionManagerClient.java [hunk 4]
```diff
@@ -116,6 +115,42 @@ private CreateInstanceResponse createInstance(final String extensionId, final St
...
+    private ExtensionInfo getExtensionInfo() {
+        final ExtensionsResponseExtension extension = getSingleExtension();
+        if (extension.getInstallableVersions().size() != 1) {
+            throw new IllegalStateException("Expected at exactly one installable version for extensions "
+                    + extension.getId() + " but got " + extension.getInstallableVersions());
+        }
+        return new ExtensionInfo(extension.getId(), extension.getInstallableVersions().get(0));
+    }
+
+    private static class ExtensionInfo {
```
Comment (ckunki) [inline src/test/java/com/exasol/adapter/document/files/extension/ExtensionManagerClient.java:136]: Would it make sense to rename this class to simply `Extension`?

**Review by ckunki — CHANGES_REQUESTED** [review]

File: extension/src/extension.ts (hunk 1 above)
Comment (kaklakariada) [reply extension/src/extension.ts:43]: Yes, correct. For now we only support a single version.
We could throw an error when someone tries to a wrong version, but that's not necessary I guess.

**Review by ckunki — APPROVED** [review]

File: src/test/java/com/exasol/adapter/document/files/extension/ExtensionManagerClient.java [hunk 5]
```diff
@@ -51,38 +51,18 @@ public List<ExtensionsResponseExtension> getExtensions() {
...
     public void installExtension() {
-        final ExtensionsResponseExtension extension = getSingleExtension();
-        if (extension.getInstallableVersions().isEmpty()) {
-            throw new IllegalStateException(
-                    "Expected at least one installable version for extensions " + extension.getId());
-        }
-        final String version = extension.getInstallableVersions().get(0);
-        LOGGER.fine(() -> "Installing extension " + extension.getId() + " in version " + version);
-        install(extension.getId(), version);
+        installExtension(extension.getCurrentVersion());
```
Comment (ckunki) [inline src/test/java/com/exasol/adapter/document/files/extension/ExtensionManagerClient.java:65]:
```suggestion
        final ExtensionInfo extension = getExtensionInfo();
        installExtension(extension.getCurrentVersion());
```

**Review by ckunki — DISMISSED** [review]

**Review by ckunki — APPROVED** [review]

**Pull request merged**